### ISSUE TITLE
Feature Request: mining latency option for hardhat network on localhost

### ISSUE THREAD

**Issue by gitpusha** [nomiclabs/hardhat#1065]
It would be amazing if there was a network config for hardhat network on localhost that lets you configure a random mining latency within a range: 

e.g. 

```javascript
    localhost: {
      latency: [15, 60],  // range in between 15 and 60 seconds
      timeout: 150000,
      forking: {
```

**Comment by fvictorio** [comment]
Related to: #1087

**Comment by fvictorio** [comment]
Oh, actually, that PR isn't *exactly* what you want, but I think it will be close enough.

**Comment by gitpusha** [comment]
> Oh, actually, that PR isn't exactly what you want, but I think it will be close enough.

Yeah, it seems to be really close. It doesnt allow to set the interval length at random out of the box though.

**Comment by fvictorio** [comment]
Hardhat [2.4.0](https://github.com/nomiclabs/hardhat/releases/tag/hardhat-core-v2.4.0) now has support for setting a min gas price such that transactions below that threshold are not mined. While that's not _exactly_ what this issue is about, I think it's enough of a building block to simulate the intended behavior.

I'm going to leave this open anyway because we might add this specific feature (send a transaction and indicate somehow that it should be included after 3 blocks) sometime in the future.

**Comment by gitpusha** [comment]
Super cool! Can't wait to try it out!
